core/primitives-core/src/global_contract.rs

```diff
@@ -0,0 +1,71 @@
+use crate::account::AccountContract;
+use crate::hash::CryptoHash;
+use borsh::{BorshDeserialize, BorshSerialize};
+use near_account_id::AccountId;
+use near_schema_checker_lib::ProtocolSchema;
+use serde_with::serde_as;
+use std::fmt;
+
+#[serde_as]
+#[derive(
+    BorshSerialize,
+    BorshDeserialize,
+    serde::Serialize,
+    serde::Deserialize,
+    Hash,
+    PartialEq,
+    Eq,
+    Clone,
+    ProtocolSchema,
+    Debug,
+)]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+#[borsh(use_discriminant = true)]
+#[repr(u8)]
+pub enum GlobalContractIdentifier {
+    CodeHash(CryptoHash) = 0,
```

Comment:
Here, too:
```suggestion
    CodeHash(#[serde_as(as = "Base64")] CryptoHash) = 0,
```

Reply:
`CryptoHash` already handles de-/serializing. It uses base58.

https://github.com/near/nearcore/blob/91e4b95619336657a40a134ba4c14aba7122da43/core/primitives-core/src/hash.rs#L137-L144

@mitinarseny  Does that resolve your concern? Or do need this to be base64?